### PR TITLE
stream-k v0.4

### PR DESCRIPTION
**Pull request by xiaohuguo2023**
- enable perf CI
- adapt tune_streamk to be able to switch kernel tunning from command line
-  add persistent_gemm unit tests
-  make test_correctness work with multiple kernels
-  add perfci tuning shapes and fall back configs

### CODE REVIEW

File: python/perf-kernels/streamk/tune_streamk.py [hunk 1]
```diff
@@ -542,6 +540,20 @@ def get_rocm_version():
...
+    kernel_name = kernel_name.strip()
+    module = importlib.import_module(module_name)
+    kernel_func = getattr(module, kernel_name)
+
+    if not args.o:
+        if args.benchmark:
+            args.o = f"benchmarking_results_{kernel_name}.csv"
+        else:
+            default_name = get_default_tuning_result_filename()
+            args.o = f"{kernel_name}_{default_name}"
```
Comment (stephen-youn) [inline python/perf-kernels/streamk/tune_streamk.py]: this line makes an error to concatenate kernel name with the path with "_", should it be just default_name?

Comment (xiaohuguo2023) [reply python/perf-kernels/streamk/tune_streamk.py]: Thanks @stephen-youn, fixed

**Comment by AlexAUT** [comment]
LGTM for the CI part 👍

File: python/perf-kernels/streamk/tune_streamk.py [hunk 2]
```diff
@@ -206,9 +205,9 @@ def profile_batch_kernels(M, N, K, gpuid, gpus, jobs, verbose):
...
     jobId = gpuIdx
     while jobId < jobs:
-        kernel_name = get_filename_profile_driver(M, N, K, jobId)
+        kernelname = get_filename_profile_driver(M, N, K, jobId)
         if verbose:
-            print(f"profiling {kernel_name} on GPU {gpuid}")
+            print(f"profiling {kernelname} on GPU {gpuid}")
         run_bash_command_wrapper(
             f"rocprof --stats -o results_{jobId}.csv python {get_filename_profile_driver(M, N, K, jobId)}",
             #            f"rocprofv2 --plugin file --plugin-version 1 --kernel-trace -o {jobId} python {get_filename_profile_driver(M, N, K, jobId)}",
```
Comment (vgokhale) [inline python/perf-kernels/streamk/tune_streamk.py:213]: Can we delete this line?

File: python/perf-kernels/streamk/tuning_gemm_sizes.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+- {'M': 512, 'N': 5120, 'K': 1024, 'rowMajorA': 'T', 'rowMajorB': 'N'}
+- {'M': 512, 'N': 5120, 'K': 1024, 'rowMajorA': 'T', 'rowMajorB': 'T'}
+- {'M': 512, 'N': 1024, 'K': 5120, 'rowMajorA': 'T', 'rowMajorB': 'N'}
+- {'M': 512, 'N': 1024, 'K': 5120, 'rowMajorA': 'T', 'rowMajorB': 'T'}
+- {'M': 2048, 'N': 5120, 'K': 512, 'rowMajorA': 'T', 'rowMajorB': 'N'}
+- {'M': 2048, 'N': 5120, 'K': 512, 'rowMajorA': 'T', 'rowMajorB': 'T'}
+
+- {'M': 20196, 'N': 512, 'K': 1536, 'rowMajorA': 'T', 'rowMajorB': 'N'}
+- {'M': 20196, 'N': 512, 'K': 1536, 'rowMajorA': 'T', 'rowMajorB': 'T'}
+- {'M': 171792, 'N': 512, 'K': 1536, 'rowMajorA': 'T', 'rowMajorB': 'N'}
```
Comment (vgokhale) [inline python/perf-kernels/streamk/tuning_gemm_sizes.yaml:77]: Just to double check - the kernel supports M dims that are not a multiple of BLOCK_M?

Comment (xiaohuguo2023) [reply python/perf-kernels/streamk/tuning_gemm_sizes.yaml:77]: yes, both streamk kernel and persistent_gemm support any sizes

File: python/perf-kernels/streamk/tune_streamk.py (hunk 2 above)
Comment (xiaohuguo2023) [reply python/perf-kernels/streamk/tune_streamk.py:213]: can we keep it here for a while, as currently rocprofv2 has bug that can't return all configs timing, if the bug is fixed, we could use rocprofv2 which may be helpful to reduce profiling costs.

**Review by neoblizz — APPROVED** [review]
LGTM

File: python/perf-kernels/streamk/tune_streamk.py (hunk 2 above)
Comment (vgokhale) [reply python/perf-kernels/streamk/tune_streamk.py:213]: Sure. May be add a comment and TODO.

**Review by vgokhale — APPROVED** [review]

**Pull request merged**